### PR TITLE
Equals and hashcode tests

### PR DESCRIPTION
**Pull request by alexander-kulikovskii**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [x] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [ ] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog></changelog>`.

### Summary of changes

In this PR added tool that allows to check manually generated `equals` and `hashCode` method. And makes sure not to forget update them after modification.

### CODE REVIEW

**Comment by Chaoba** [comment]
Shall we add a check for class `Plugin`?

File: buildSrc/src/main/kotlin/Project.kt [hunk 1]
```diff
@@ -63,6 +63,8 @@ object Dependencies {
   const val testUtils = "com.android.tools:testutils:${Versions.lint}"
   const val hamcrest = "org.hamcrest:hamcrest:${Versions.hamcrest}"
   const val annotations = androidxAnnotations
+  const val equalsVerifier = "nl.jqno.equalsverifier:equalsverifier:${Versions.equalsVerifier}"
+
```
Comment (kiryldz) [inline buildSrc/src/main/kotlin/Project.kt:67]: empty line

File: extension-style/build.gradle.kts [hunk 2]
```diff
@@ -27,6 +27,7 @@ dependencies {
   testImplementation(Dependencies.mockk)
   testImplementation(Dependencies.androidxTestCore)
   testImplementation(Dependencies.robolectric)
+  testImplementation(Dependencies.equalsVerifier)
```
Comment (kiryldz) [inline extension-style/build.gradle.kts:30]: it depends on `sdk-base` where we also specify this dependency, guess we do not need it here then

**Review by kiryldz — APPROVED** [review]
LGTM with couple nits.
Also discussed offline that some follow-up work may be done, please create follow-up ticket for it.

File: extension-style/build.gradle.kts (hunk 2 above)
Comment (alexander-kulikovskii) [reply extension-style/build.gradle.kts:30]: it's not like `testApi`

I want to refactor a little bit test dependencies

File: module-telemetry/build.gradle.kts [hunk 3]
```diff
@@ -32,6 +32,7 @@ dependencies {
   testImplementation(Dependencies.junit)
   testImplementation(Dependencies.mockk)
   testImplementation(Dependencies.robolectric)
+  testImplementation(Dependencies.equalsVerifier)
```
[the remark this answers is not in the document]
Comment (alexander-kulikovskii) [reply module-telemetry/build.gradle.kts:35]: same as above. leave as is

**Review by pengdev — APPROVED** [review]
LGTM :shipit:

**Review by alexander-kulikovskii — COMMENTED** [review]
> Shall we add a check for class `Plugin`?

good point. miss this class

File: buildSrc/src/main/kotlin/Project.kt (hunk 1 above)
Comment (alexander-kulikovskii) [reply buildSrc/src/main/kotlin/Project.kt]: fix it

**Pull request merged**